e2e-tests/src/test/java/dev/streamx/cli/StreamxCliTestsIT.java

```diff
@@ -0,0 +1,143 @@
+package dev.streamx.cli;
+
+
+import dev.streamx.cli.test.tools.ResourcePathResolver;
+import dev.streamx.cli.test.tools.terminal.TerminalCommandRunner;
+import dev.streamx.cli.test.tools.validators.HttpValidator;
+import dev.streamx.cli.test.tools.validators.ProcessValidator;
+import io.quarkus.test.junit.QuarkusTest;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class StreamxCliTestsIT {
+
+  @ConfigProperty(name = "web.delivery.port.url", defaultValue = "http://localhost:8081/")
+  String webDeliveryPortUrl;
+
+  @Inject
+  @Named("StreamxCommandRunner")
+  TerminalCommandRunner terminalCommandRunner;
+
+  @Inject
+  ProcessValidator processValidator;
+
+  @Inject
+  ResourcePathResolver resourcePathResolver;
+
+  @Inject
+  HttpValidator httpValidator;
+
+  @BeforeAll
+  public void setup() {
+    streamxCommand(
+        "run -f " + filePath("streamx-mesh.yml"),
+        "STREAMX IS READY!",
+        60);
```

Comment:
expose this timeout as @ConfigProperty - maybe prefixed with "streamx.cli.e2e.setup.timeout"